modules/build/src/test/scala/scala/build/tests/BuildTests.scala

```diff
@@ -917,11 +918,32 @@ abstract class BuildTests(server: Boolean) extends munit.FunSuite {
           |}
           |""".stripMargin
     )
-    inputs.withBuild(baseOptions, buildThreads, bloopConfigOpt) { (_, _, maybeBuild) =>
-      expect(maybeBuild.exists(_.success))
-      val build = maybeBuild.toOption.flatMap(_.successfulOpt).getOrElse(sys.error("cannot happen"))
-      val cp    = build.fullClassPath
-      expect(cp.length == 1) // no scala-library, only the class directory
+    val options = baseOptions.copy(
+      scalaOptions = baseOptions.scalaOptions.copy(
+        supportedScalaVersionsUrl =
+          Some("abcd://nope"), // should not be checked, as we're a pure Java project
+        ignoreSupportedScalaVersionsErrors = Some(false)
+      )
+    )
+    val gotExpectedError =
+      try {
+        options.scalaParams
+        false
+      }
+      catch {
+        case e if e.getCause.isInstanceOf[ArtifactError.DownloadError] =>
```

Comment:
If we want to avoid downloading supported Scala Version, why not change type `ScalaOptions.scalaVersionsUrl` to `Option[String]`. And download supportedScalaVersion only when variable `supportedScalaVersionsUrl ` have some value.

Instead of catching a download error, we just might not download the supported Scala Versions.

Reply:
I don't think that would work for this PR. Here, I want the supported Scala version check to fail loudly, so that the test just below those lines only passes if it doesn't rely at all on supported Scala version stuff.

We can't make `ScalaOptions.scalaVersionsUrl` have a default value like `Some("https://…")`, as that would make some `BuildOptions` calculations inconsistent (like `myOptions orElse BuildOptions()` would erase a `ScalaOptions.scalaVersionsUrl` set to `None` in `myOptions`, while we want that to be equal to just `myOptions` - this "breaks the monoid laws"). So we can't rely on `ScalaOptions.scalaVersionsUrl` to be `None` here.

We could set it to `Some("")` to disable downloading a listing, but then the most sensible thing to do would be to rely on the [hardcoded version list](https://github.com/VirtusLab/scala-cli/blob/673239d8c29d7152fca675061190bd05b010a991/modules/options/src/main/scala/scala/build/options/BuildOptions.scala#L263) like we do already when downloading the listing fails.

So a separate field for this is better I think.